src/box-node-sdk.ts

```diff
@@ -0,0 +1,463 @@
+	 * the API.
+	 *
+	 * @returns {BoxClient} Returns a new Box Client paired to a AnonymousAPISession
+	 */
+	getAnonymousClient() {
+		return new BoxClient(
+			this.anonymousSession,
+			this.config,
+			this.requestManager
+		);
+	}
+
+	/**
+	 * Create a new client using App Auth for the given entity.  This allows either
+	 * managing App Users (as the enterprise) or performing operations as the App
+	 * Users themselves (as a user).
+	 *
+	 * @param {string} type The type of entity to operate as, "enterprise" or "user"
+	 * @param {string} id The Box ID of the entity to operate as
+	 * @param {TokenStore} [tokenStore] (Optional) the token store to use for caching tokens
+	 * @returns {BoxClient} A new client authorized as the app user or enterprise
+	 */
+	getAppAuthClient(type: string, id: string, tokenStore?: TokenStore) {
+		if (type === 'enterprise' && !id) {
+			if (this.config.enterpriseID) {
+				id = this.config.enterpriseID;
+			} else {
+				throw new Error('Enterprise ID must be passed');
+			}
+		}
+
+		var appAuthSession = new AppAuthSession(
+			type,
+			id,
+			this.config,
+			this.tokenManager,
+			tokenStore
+		);
+		return new BoxClient(appAuthSession, this.config, this.requestManager);
+	}
+
+	/**
+	 * Generate the URL for the authorize page to send users to for the first leg of
+	 * the OAuth2 flow.
+	 *
+	 * @param {Object} params The OAuth2 parameters
+	 * @returns {string} The authorize page URL
+	 */
+	getAuthorizeURL(params: { client_id?: string }) {
+		params.client_id = this.config.clientID;
+
+		return `${this.config.authorizeRootURL}/oauth2/authorize?${qs.stringify(
+			params
+		)}`;
+	}
+
+	/**
+	 * Acquires token info using an authorization code
+	 *
+	 * @param {string} authorizationCode - authorization code issued by Box
+	 * @param {TokenRequestOptions} [options] - Sets optional behavior for the token grant, null for default behavior
+	 * @param {Function} [callback] - passed a TokenInfo object if tokens were granted successfully
+	 * @returns {Promise<TokenInfo>} Promise resolving to the token info
+	 */
+	getTokensAuthorizationCodeGrant(
+		authorizationCode: string,
+		options: TokenRequestOptions | null,
+		callback: Function
+	) {
+		return this.tokenManager
+			.getTokensAuthorizationCodeGrant(
+				authorizationCode,
+				options as any /* FIXME */
+			)
+			.asCallback(callback);
+	}
+
+	/**
+	 * Refreshes the access and refresh tokens for a given refresh token.
+	 *
+	 * @param {string} refreshToken - A valid OAuth refresh token
+	 * @param {TokenRequestOptions} [options] - Sets optional behavior for the token grant, null for default behavior
+	 * @param {Function} [callback] - passed a TokenInfo object if tokens were granted successfully
+	 * @returns {Promise<TokenInfo>} Promise resolving to the token info
+	 */
+	getTokensRefreshGrant(
+		refreshToken: string,
+		options: TokenRequestOptions | null,
```

Comment:
should this be `options: TokenRequestOptions | Function | null,` based on line 366